### ISSUE TITLE
the "source" returned by "originalPositionFor" is different with "sources" of sourceMap file

### ISSUE THREAD

**Issue by kiwenlau** [mozilla/source-map#267]
I use "originalPositionFor" to get the original position. 

```js
    console.log(JSON.stringify(sourceMap.sources, null, 4));

    var smc = new SourceMap.SourceMapConsumer(sourceMap);
    var result = smc.originalPositionFor(
    {
        line: parseInt(lineNumber),
        column: parseInt(columnNumber)
    });

    console.log(result.source);
```

I print out the "sources" property of sourceMap and the "source" of "originalPosition" results, but they are different, making it hard for me to locate the exact "sourcesContent" of sourceMap.

sourceMap.sources:
```js
[
    "webpack:///static/js/manifest.b7cf97680f7a50fa150f.js",
    "webpack:///webpack/bootstrap 67e184f9679733298d44"
]
```

result.source:
```js
webpack:/webpack/bootstrap 67e184f9679733298d44
```

result.source misses "//", which is quite strange.

Any suggestion? Thx!

**Comment by tromey** [comment]
Source maps for webpack-built bundles are definitely working in Firefox.  For example, [this test](https://dxr.mozilla.org/mozilla-central/rev/a49112c7a5765802096b3fc298069b9495436107/devtools/client/framework/test/browser_source_map-reload.js#1) uses one.

The best thing would be if you could make a (hopefully minimal) example showing the problem.

**Comment by kiwenlau** [comment]
Here is the Source Map file:(GitHub only support uploading .txt file...)
[test.txt](https://github.com/mozilla/source-map/files/1063085/test.txt)

The sources field of source map:

```
"sources": ["webpack:///static/js/manifest.b7cf97680f7a50fa150f.js","webpack:///webpack/bootstrap 67e184f9679733298d44"],
```

Here is the code:

```js
const sourceMap = require('source-map');

const fs = require('fs');
const rawSourceMap = JSON.parse(fs.readFileSync('./test.txt', 'utf8'));

const smc = new sourceMap.SourceMapConsumer(rawSourceMap);

console.log(smc.originalPositionFor({
  line: 1,
  column: 100
}));
```

Here is the output:
```
{ source: 'webpack:/webpack/bootstrap 67e184f9679733298d44',
  line: 52,
  column: 0,
  name: 'call' }
```

You can see that **the "source" returned by "originalPositionFor" is different with "sources" of sourceMap file**

Thx!

**Comment by kiwenlau** [comment]
I found exactly the same problem again! @tromey 

here is the sourcemap file: [test02.txt](https://github.com/mozilla/source-map/files/1230546/test02.txt)

The sources field of source map:

```javascript
"webpack:///./src/components/ajax/ajax.js"
```

Here is the code:

```javascript
const sourceMap = require('source-map');

const fs = require('fs');
const rawSourceMap = JSON.parse(fs.readFileSync('./test02.txt', 'utf8'));

const smc = new sourceMap.SourceMapConsumer(rawSourceMap);

console.log(smc.originalPositionFor({
  line: 1,
  column: 15545
}));
```

Here is the output:

```javascript
{ source: 'webpack:///src/components/ajax/ajax.js',
  line: 14,
  column: 23,
  name: 'data' }
```

**Comment by tromey** [comment]
I would mildly expect the `///` to be transformed to `//` by source-map's URL normalization; but I definitely am seeing it transform to `/`, which seems quite wrong.

**Comment by tromey** [comment]
The built-in URL parser can't parse that URL, which is probably the problem.  Would be fixed by using the standard URL, see #275

**Comment by tromey** [comment]
> { source: 'webpack:///src/components/ajax/ajax.js',

This one isn't really a bug.  The source-map library does some normalization.
